tools/bin/fetch_docs.js

```diff
@@ -0,0 +1,144 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+"use strict";
+
+var fs            = require("fs");
+var fse           = require("fs-extra");
+var https         = require("https");
+var path          = require("path");
+var child_process = require("child_process");
+var yaml          = require("js-yaml");
+
+// constants
+var DEFAULT_REPO_PATH = "README.md";
+
+function generateFrontMatter (useDirectSrcURI, filePath, name, version) {
+    var obj                 = {};
+
+    if(useDirectSrcURI) {
+        filePath            = filePath.replace("https://raw.githubusercontent.com/", "");
+        var filePathSplit   = filePath.split("/");
+        name                = filePathSplit[0] + "/" + filePathSplit[1];
+        version             = filePathSplit[2];
+        filePath            = filePathSplit.slice(3).join("/");
+    }
+
+    obj.edit_link           = getRepoURIForFrontMatter(name, version, filePath);
+    obj.note_link           = "https://github.com/" + name + "/releases";
+    obj.note                = "This documentation is for the " + version + " version of the " + name + ".";
+    obj.note_link_text      = "Visit Github for other versions.";
```

Comment:
Please move this logic to the Jekyll side. Instead set a variable called something like `plugin_version`, and construct the link in HTML.